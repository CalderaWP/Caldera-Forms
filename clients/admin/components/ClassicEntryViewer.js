import {Component} from 'react';
import PropTypes from 'prop-types';


export const ClassicEntryViewer = (props) => (
	<div>
		<button onClick={props.onClose}>Close</button>
		<div>I am the entry viewer for {props.form.name}</div>
	</div>
);
