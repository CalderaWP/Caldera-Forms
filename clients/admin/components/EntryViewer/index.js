import {EntryViewer} from "./EntryViewer";
import {EntryRowActions} from "./EntryRowActions";
import {Entry} from "./Entry";
import {FormEntryViewer} from "./FormEntryViewer";
import entryType from './entryType';
import formType from './formType';
export default {
	EntryViewer,
	EntryRowActions,
	Entry,
	FormEntryViewer,
	entryType,
	formType
};