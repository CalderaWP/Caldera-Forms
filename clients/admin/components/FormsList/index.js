import {Form} from "./Form";
import {FormList} from "./FormList";
import {ShortcodeViewer} from "./ShortcodeViewer";

/**
 * Export all FormList related components
 */
export default {
	Form,
	FormList,
	ShortcodeViewer
}
