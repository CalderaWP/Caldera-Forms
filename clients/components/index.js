
export {
    RemotePost,
    FeaturedImage,
    Image
} from './RemotePost';


export {
    DocSearch,
    DocSearchApp
} from './DocSearch'

export {
    Catdera
} from './Catdera/Catdera';