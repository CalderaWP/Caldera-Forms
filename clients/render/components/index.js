import {CalderaFormsFieldRender,CalderaFormsFieldPropType} from "./CalderaFormsFieldRender";
import {CalderaFormsRender} from "./CalderaFormsRender";
import {CalderaFormsFieldGroup} from "./CalderaFormsFieldGroup";

/**
 * The Caldera Forms version 2 form renderer.
 *
 * @since 1.8.0
 */
export{
	CalderaFormsFieldGroup,
	CalderaFormsFieldPropType,
	CalderaFormsFieldRender,
	CalderaFormsRender
}
