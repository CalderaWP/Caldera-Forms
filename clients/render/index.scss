@import 'components/Fields/styles/files-field';