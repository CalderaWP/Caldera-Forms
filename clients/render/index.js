/** globals CF_API_DATA **/
import './index.scss';
import {CalderaFormsRender} from "./components/CalderaFormsRender";

import domReady from '@wordpress/dom-ready';
import {
	hashFile,
	createMediaFromFile,
	captureRenderComponentRef
} from "./util";
import { handleFileUploadResponse, handleFileUploadError, hashAndUpload, processFiles, processFileField, processFormSubmit } from './fileUploads';

domReady(function () {
	jQuery(document).on('cf.form.init', (e, obj) => {
		const {
			state,//CF state instance
			formId, //Form's ID
			idAttr, //Form element id attribute,
			//$form, //Form jQuery object
		} = obj;
		const fieldsToControl = [];
		if( 'object' !== typeof  window.cf2 ){
			window.cf2 = {};
		}


		//Build configurations
		document.querySelectorAll('.cf2-field-wrapper').forEach(function (fieldElement) {
			const fieldIdAttr = fieldElement.getAttribute('data-field-id');

			const formConfig = window.cf2[idAttr];

			let fieldConfig = formConfig.fields.hasOwnProperty(fieldIdAttr) ?
				formConfig.fields[fieldIdAttr]
				: null;

			if ('string' === typeof  fieldConfig) {
				fieldConfig = JSON.parse(fieldConfig);
			}
			if (fieldConfig) {
				fieldsToControl.push(fieldConfig);
				if (fieldConfig.hasOwnProperty('fieldDefault')) {
					state.mutateState(fieldIdAttr, fieldConfig.fieldDefault);

				}
			}
		});



		/**
		 * Flag to indicate if validation is happening or not
		 *
		 * This is controlled outside of React so that CF1 can trigger validation
		 *
		 * @since 1.8.0
		 *
		 * @type {boolean}
		 */
		let shouldBeValidating = false;
		let messages = {};


		jQuery(document).on('cf.ajax.request', (event, obj) => {
			//Compare the event form id with the component form id
			if(!obj.hasOwnProperty('formIdAttr') || obj.formIdAttr !== idAttr){
				return;
			}
			shouldBeValidating = true;
			const values = theComponent.getAllFieldValues();
			const cf2 = window.cf2[obj.formIdAttr];
			if(typeof cf2 !== "undefined" && cf2.length > 0){
				cf2.formIdAttr = obj.formIdAttr;
			}
			const {displayFieldErrors,$notice,$form,fieldsBlocking} = obj;
			if ('object' !== typeof cf2) {
				return;
			}

			cf2.pending = cf2.pending || [];
			cf2.uploadStarted = cf2.uploadStarted || [];
			cf2.uploadCompleted = cf2.uploadCompleted || [];
			cf2.fieldsBlocking = cf2.fieldsBlocking || [];


			if (Object.keys(values).length) {
				Object.keys(values).forEach(fieldId => {
					const field = fieldsToControl.find(field => fieldId === field.fieldId);
					if (field) {
						if ('file' === field.type) {
							const processFunctions = {processFiles, hashAndUpload, hashFile, createMediaFromFile, handleFileUploadResponse, handleFileUploadError, processFormSubmit};
							const processData = {obj, values, field, fieldId, cf2, $form, CF_API_DATA, theComponent, messages};
							processFileField(processData, processFunctions);
						}
					}
				});
			} else {
				if(typeof field !== "undefined"){
					obj.$form.data(field.fieldId, values[field.fieldId]);
				}
			}

		});

		/**
		 * Ref for rendered app
		 *
		 * @see https://reactjs.org/docs/refs-and-the-dom.html
		 *
		 * @since 1.8.0
		 * @type {*}
		 */
		let theComponent = '';
		wp.element.render(
			<CalderaFormsRender
				cfState={state}
				formId={formId}
				formIdAttr={idAttr}
				fieldsToControl={fieldsToControl}
				shouldBeValidating={shouldBeValidating}
				ref={(component) => {
					captureRenderComponentRef(component,idAttr,window);
					theComponent = component
				}}
				messages={messages}
				strings={CF_API_DATA.strings}
			/>,
			document.getElementById(`cf2-${idAttr}`)
		);

	});

});


