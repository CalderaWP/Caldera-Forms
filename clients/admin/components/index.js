import EntryViewer from './EntryViewer/index';
import FormsList from './FormsList/index';
import NewForm from './NewForm/index';
import RemotePost from './RemotePost/index';
import util from './util/index';

/**
 *
 */
export default {
	EntryViewer,
	FormsList,
	NewForm,
	RemotePost,
	Layout,
	util
};