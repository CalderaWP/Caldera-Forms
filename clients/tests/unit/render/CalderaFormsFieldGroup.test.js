import {CalderaFormsFieldGroup} from "../../../render/components";
import renderer from 'react-test-renderer';
const handler = () => {
};
import {shallow, mount} from 'enzyme';
import EnzymeAdapter from '../createEnzymeAdapter'


describe('CalderaFormsFieldGroup component', () => {
	const fieldIdAttr = 'fld_5843941_1';
	const fieldConfig = {
		fieldCaption: "",
		fieldDefault: "new default",
		fieldId: "fld_5843941",
		fieldIdAttr,
		fieldLabel: "Text 2 Field",
		fieldPlaceHolder: "",
		fieldValue: "",
		outterIdAttr: "cf2-fld_5843941_1",
		required: false,
		type: "text",
		areValuesValid: true
	};

	const getFieldConfig = (fieldIdAttr) =>{
		return fieldConfig;
	};
	it('Matches snapshot with basic args', () => {
		expect(
			renderer.create(
				<CalderaFormsFieldGroup
					onChange={handler}
					field={fieldConfig}
					getFieldConfig={getFieldConfig}
				/>
			).toJSON()
		).toMatchSnapshot();
	});


	it('Does not render if shouldShow is false', () => {
		expect(
			renderer.create(
				<CalderaFormsFieldGroup
					onChange={handler}
					field={fieldConfig}
					shouldShow={false}
					getFieldConfig={getFieldConfig}

				/>
			).toJSON()
		).toEqual(null)
	});

	it('Creates the label correctly', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={fieldConfig}
			shouldDiable={true}
			getFieldConfig={getFieldConfig}
		/>);
		const testInstance = testRenderer.root;

		expect(testInstance.findByType('label').props.htmlFor).toBe(fieldIdAttr);
		expect(testInstance.findByType('label').props.className).toBe('control-label');


	});

	it('Creates the label with the right text', () => {
		const testRenderer = mount(<CalderaFormsFieldGroup
			onChange={handler}
			field={fieldConfig}
			shouldDiable={true}
			getFieldConfig={getFieldConfig}
		/>);
		const testInstance = testRenderer.root;


		expect(testRenderer.contains(fieldConfig.fieldLabel) ).toBe( true );


	});


	it('Creates the inner input correctly', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={fieldConfig}
			getFieldConfig={getFieldConfig}
		/>);
		const testInstance = testRenderer.root;

		expect(testInstance.findByType('input').props.required).toBe(false);
		expect(testInstance.findByType('input').props.className).toBe('cf2-text form-control');
		expect(testInstance.findByType('input').props.placeholder).toBe('');
		expect(testInstance.findByType('input').props.type).toBe('text');
		expect(testInstance.findByType('input').props.value).toBe('');
		expect(typeof testInstance.findByType('input').props.onChange).toBe('function');


	});

	it('Disables is shouldDisable is true', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={fieldConfig}
			shouldDisable={true}
			getFieldConfig={getFieldConfig}
		/>);
		const testInstance = testRenderer.root;

		expect(testInstance.findByType('input').props.disabled).toBe(true);

	});

	it('Sets required if it should', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={{
				...fieldConfig,
				required: true
			}}
			getFieldConfig={getFieldConfig}

		/>);
		const testInstance = testRenderer.root;

		expect(testInstance.findByType('input').props.required).toBe(true);

	});

	it('Adds caption and aria attribute', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={{
				...fieldConfig,
				caption: "Click Me"
			}}
		/>);
		const testInstance = testRenderer.root;
		expect(testInstance.findByType('span').props.className).toBe('help-block');
		expect(testInstance.findByType('span').props.children).toBe('Click Me');

	});

});

describe('chooses inner field', () => {
	const fileFieldConfig = {
		field:
			{
				type: 'file',
				outterIdAttr: 'cf2-not_required_multiple_has_button_text',
				fieldId: 'not_required_multiple_has_button_text',
				fieldLabel: 'Not Required Multiple Has Button Text',
				fieldCaption: '',
				fieldPlaceHolder: '',
				required: 'false',
				fieldDefault: '',
				fieldValue: '',
				fieldIdAttr: 'not_required_multiple_has_button_text',
				areValuesValid: true
			},
		accept: '',
		multiple: 'false',
		multiUploadText: 'false'
	};

	const textFieldConfig = {
		fieldCaption: "",
		fieldDefault: "new default",
		fieldId: "cf2-fdl1",
		fieldIdAttr: "fdl1_1",
		fieldLabel: "Text 2 Field",
		fieldPlaceHolder: "",
		fieldValue: "",
		outterIdAttr: "cf2-fdl1_1",
		required: false,
		type: "text",
	};

  const strings = {
    cf2FileField: {
      removeFile: 'Remove file',
      defaultButtonText: 'Drop files or click to select files to Upload'
    }
  }

	const getFieldConfig = (fieldIdAttr) =>{
		return textFieldConfig;
	};
	it( 'uses text input for text field', () =>{
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={textFieldConfig}
			getFieldConfig={getFieldConfig}
		/>);
		const testInstance = testRenderer.root;
		expect(testInstance.findByType('input').props.type).toBe('text');
		expect(testInstance.findByType('input').props.className).toBe('cf2-text form-control');

	});

	it( 'uses file input for text field', () =>{
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={fileFieldConfig.field}
			getFieldConfig={getFieldConfig}
			strings={strings.cf2FileField}
		/>);
		const testInstance = testRenderer.root;
		expect(testInstance.findByType('input').props.type).toBe('file');
		expect(testInstance.findByType('div').props.className).toBe('form-group cf2-field-group');
	});

});

describe( 'DOM testing of CalderaFormsFieldGroup component', () => {
	const textFieldConfig = {
		fieldCaption: "",
		fieldDefault: "new default",
		fieldId: "cf2-fdl1",
		fieldIdAttr: "fdl1_1",
		fieldLabel: "Text 2 Field",
		fieldPlaceHolder: "",
		fieldValue: "",
		outterIdAttr: "cf2-fdl1_1",
		required: false,
		type: "text",
		isRequired: true
	};

	it('Shows required indicator when it should', () => {
		const component = mount(<CalderaFormsFieldGroup
			onChange={handler}
			field={textFieldConfig}
			shouldDisable={true}
			getFieldConfig={() => {}}
		/>);

		expect(component.contains('*')).toBe(true);
		expect(component.find('.field_required').length).toBe(1);

	});
});
