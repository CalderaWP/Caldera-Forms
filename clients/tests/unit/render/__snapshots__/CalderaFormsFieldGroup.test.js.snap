// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalderaFormsFieldGroup component Matches snapshot with basic args 1`] = `
<div
  className="form-group cf2-field-group"
>
  <label
    className="control-label"
    htmlFor="fld_5843941_1"
    id="fld_5843941_1Label"
  >
    Text 2 Field
  </label>
  <input
    aria-describedby="fld_5843941_1Caption"
    className="cf2-text form-control"
    data-field="fld_5843941"
    data-type="cf2-text"
    disabled={false}
    id="fld_5843941_1"
    name="fld_5843941"
    onChange={[Function]}
    placeholder=""
    required={false}
    type="text"
    value=""
  />
</div>
`;
